src/Codeception/Scenario.php

```diff
@@ -117,54 +127,46 @@ public function getHtml()
             }
         }
         $text = str_replace(['"\'', '\'"'], ["'", "'"], $text);
-        $text = "<h3>" . mb_strtoupper('I want to ' . $this->getFeature(), 'utf-8') . "</h3>" . $text;
-        return $text;
+        return "<h3>" . mb_strtoupper('I want to ' . $this->getFeature(), 'utf-8') . "</h3>" . $text;
     }
 
-    public function getText()
+    public function getText(): string
     {
         $text = '';
         foreach ($this->getSteps() as $step) {
-            $text .= $step->getPrefix() . "$step \r\n";
+            $text .= $step->getPrefix() . "{$step} \r\n";
         }
         $text = trim(str_replace(['"\'', '\'"'], ["'", "'"], $text));
-        $text = mb_strtoupper('I want to ' . $this->getFeature(), 'utf-8') . "\r\n\r\n" . $text . "\r\n\r\n";
-        return $text;
+        return mb_strtoupper('I want to ' . $this->getFeature(), 'utf-8') . "\r\n\r\n" . $text . "\r\n\r\n";
     }
 
-    public function comment($comment)
+    public function comment(?string $comment): void
```

Comment:
Gherkin is passing `null` again, it would be better to make it accept strings only and update `Codeception\Test\Gherkin::runStep` to pass an empty string instead.